### PR TITLE
[WIP] Add inpainting training script

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

Add in-painting training script.

### CODE REVIEW

**Review by hackkhai — APPROVED** [review]

**Comment by KaustubhPatange** [comment]
Gave this script a try it seems like it cannot be used for fine-tuning an existing sdxl inpaint model and can only be used as creating an inpainting model from base sdxl. 

The error says that the number of input channels in the unet required should be 9 (base sdxl) but is 19 (inpainting sdxl). I don't know if it is possible to scale the input tensor channel from 9 to 19 since the input already contains noisy latents + mask + mask_latents.

**Comment by AlekseyKorshuk** [comment]
Hey @patil-suraj, thanks for sharing this training code. 

Could you please share with us the exact training args (without dataset sharing) you used to train [inpainting model](https://huggingface.co/diffusers/stable-diffusion-xl-1.0-inpainting-0.1) including number of GPUs used (to get total batch size)?
Also would appreciate sharing wandb run if possible 🤗
